en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/deploy-api/deploy-graphql-api-in-choreo-connect.md

```diff
@@ -0,0 +1,293 @@
+    - GraphQL schema file (SDL) and a GraphQL server implementaion relevant to the SDL.
+
+### Step 1 - Create a GraphQL API in WSO2 API Manager publisher portal
+
+You can create a GraphQL API in WSO2 API-M publisher portal by following the steps described below.
+
+
+1. Sign in to the API Publisher Portal.
+
+    `https://<hostname>:9443/publisher`
+
+    Example: `https://localhost:9443/publisher`
+
+    Let's use `admin` as the username and password to sign in.
+
+2. Click **Create API** and then click **Import GraphQL SDL**.
+
+    [![Create GraphQL Schema Option]({{base_path}}/assets/img/learn/create-graphql-schema-option.png)]({{base_path}}/assets/img/learn/create-graphql-schema-option.png)
+
+3. Import the schema by dragging and dropping the file or by uploading the file, and click **Next**.
+
+    Let's use the [StarWarsAPI schema definition](../../assets/attachments/learn/schema_graphql.graphql) to create the schema file. 
+
+    <div class="admonition note">
+    <p class="admonition-title">Note</p>
+    <ul><li>
+        <p>You need to define the SDL Schema based on the [GraphQL schema design best practices](https://leapgraph.com/graphql-schema-design-best-practices).</p></li>
+        <li>The file extension can be either `.graphql`, `.txt`, or `.json`. </li><li> The file name can be any name, which is based on your preference.</li></ul>
+    </div>
+
+    [![Import a GraphQL schema by adding a file]({{base_path}}/assets/img/learn/import-graphql-schema-via-file.png){: style="width:80%"}]({{base_path}}/assets/img/learn/import-graphql-schema-via-file.png)
+
+4. Enter the GraphQL API related details and click **Create**.
+
+    Let's create an API named "StarWarsAPI" using the following sample data.
+
+       <table>
+          <thead>
+            <tr class="header">
+                <th><div>
+                    <div>
+                        <b>Field</b>
+                    </div>
+                    </div>
+                </th>
+          <th><div>
+          <div>
+          <b>Description</b>
+          </div>
+          </div></th>
+          </tr>
+          </thead>
+          <td >
+             <p>Name</p>
+          </td>
+          <td>
+             <p>StarWarsAPI</p>
+          </td>
+          </tr>
+          <tr>
+          <td>
+             <p>Context</p>
+          </td>
+          <td>
+             <p><code>/swapi</code></p>
+          </td>
+          </tr>
+          <tr>
+          <td>
+             <p>Version</p>
+          </td>
+          <td>
+             <p>1.0.0</p>
+          </td>
+          </tr>
+          <tr>
+          <td>
+             <p>Endpoint</p>
+          </td>
+          <td>
+             <a href="http://localhost:8080/graphql" target="_blank">http://localhost:8080/graphql</a>
```

Comment:
Shall give something like http://backend:8080/graphql and mention the etc mapping here?